Core/include/Acts/Utilities/ParameterDefinitions.hpp

```diff
@@ -235,6 +235,65 @@ template <>
 struct BoundParameterTraits<BoundParametersIndices::eBoundTime> {
   using type = unbound_parameter;
 };
+
+template <FreeParametersIndices>
+struct FreeParameterTraits;
+template <>
+struct FreeParameterTraits<FreeParametersIndices::eFreePos0> {
+  using type = unbound_parameter;
+};
+template <>
+struct FreeParameterTraits<FreeParametersIndices::eFreePos1> {
+  using type = unbound_parameter;
+};
+template <>
+struct FreeParameterTraits<FreeParametersIndices::eFreePos2> {
+  using type = unbound_parameter;
+};
+template <>
+struct FreeParameterTraits<FreeParametersIndices::eFreeTime> {
+  using type = unbound_parameter;
+};
+template <>
+struct FreeParameterTraits<FreeParametersIndices::eFreeDir0> {
+  static constexpr double pMin() { return -1.; }
```

Comment:
I am not sure if bounding the direction components is the right thing to do, since the direction vector does not has to be normalized. It usually is, but I can easily image to use the magnitude to store one additional parameter.

Reply:
If one uses a direction vector then it is expected that it is normalised. If you want to store other information in this vector than the vector is no longer a direction vector and should be declared in a different set of parameters.
Restricting the components itself to an interval [-1,1] is not the perfect solution as the correct way would be to constrain the length of all 3 components to one but it this would (if at least possible) require a lot of code around it.

Reply:
I disagree about the naming here. Saying something is a direction vector means that I am not interested in its magnitude. A momentum vector is a perfectly fine direction vector. Whether it is normalized or not is not relevant to the information it encodes.

In most use cases, we enforce normalization because it makes our life easier. In those cases, we have a *unit direction* vector.

So, restricting this to [-1,1] might break some valid use cases. What are we gaining from doing so here?

Reply:
If you use a momentum vector then it is a direction vector multiplied by the absolute value of your momentum. This is the convention we use and since the absolute value of the momentum is stored as a separate parameter, the information is already present.
Actually this parameter is assumed to be a normalised vector wherever it is used. Restricting it to reasonable values follows the same pattern as we do it for e.g. angles.

Reply:
> If you use a momentum vector then it is a direction vector multiplied by the absolute value of your momentum.

Yes, of course, but that is not the point. I may have created a bit of confusion with my original point that you could store one additional parameter in magnitude. Even if you store just the direction, you are encoding two parameters (e.g. angles) in three components (i.e. direction vector). By definition, the magnitude is irrelevant to the parameters stored. However, depending on your track model you might want to enforce a certain normalization to make your life easier. That does not mean that it always has to be normalized to unity. E.g. in a linear track model you have the freedom to choose the normalization to define how you measure propagation length, i.e. you have `\vec{r} = \vec{r}_0 + l \vec{D}`.

This is different for angles. They are bound by math, not by convention. Regardless of the track model or their interpretation they are always bound to the same interval. This is not the case for the direction vector.

Reply:
Considering the angles, there is actually no real boundary as they are periodic. But that is just a detail.
You are right about your norm point. In my understanding we enforce a certain parametrisation with corresponding conventions by offering the given enums. If a different parametrisation is more convenient for a certain application then it should become a new set of parameters with different constraints. This is actually supported e.g. in the modifications of this PR. Within the context of the `FreeParametersIndices` we always believe that the direction vector is normalised to unity although we don't enforce it by any constraints.

Reply:
Right, I think while I agree that the direction vector should be normalized, but this bounding here doesn't actually do that but something else. So I guess it makes the most sense to remove the bounding here.

Reply:
That it does not actually enforce a normalised vector I mentioned above. However I tried to have at least some restriction on the components and I don't know whether it is possible to check the whole vector in this way

Reply:
I don't see how an 'arbitrary' wrong check is better than no check.

Reply:
Why is it wrong?

Reply:
Bounding the parameters individually to `[-1, 1]` does not make `(dir0, dir1, dir2)` a normalized vector.

Reply:
But this is what I meant. Nevertheless these conditions still need to be fulfilled and therefore I don't see why it is wrong, it's just too weak.

Reply:
Removed